developers/wallets/exchanges.md

```diff
@@ -0,0 +1,188 @@
+---
+layout: subpage
+title: "Handling a VC API Exchange in a Web Wallet"
+permalink: /developers/wallets/exchanges/
+---
+
+## {{ title }}
+
+> A VC API workflow defines a particular set of steps for exchanging verifiable
+> credentials between two parties across a trust boundary. Each step can involve
+> the issuance, verification, transmission, or presentation of verifiable
+> credentials.
+
+A VC API workflow interaction begins with _either_ a CHAPI event or an
+interaction URL via a QR code (or similar user initiated transfer experience).
+
+The CHAPI event will provide a complete protocols object similar to the
+following:
+```json
+{
+  "protocols": {
+    "vcapi": "https://vcapi.example.com/workflows/abc/exchanges/123",
+    "OIC4VCI": "openid-credential-offer://?..."
+  }
+}
+```
+Alternatively, a QR code can be used to provide a URL which when dereferenced
+will result in either a `protocols` (object as above) or an HTML fallback page
+to allow the user to continue otherwise.
+
+To retrieve a `protocols` object from an _interaction URL_, the Wallet must send
+an HTTP GET request including an explicit `Accept: application/json` request
+header--which results in the same JSON object as above:
+
+```http
+GET /workflows/abc/exchanges/123/protocols
+Host: vcapi.example.com
+Accept: application/json
+
+{
+  "protocols": {
+    "vcapi": "https://vcapi.example.com/workflows/abc/exchanges/123",
+    "OIC4VCI": "openid-credential-offer://?..."
+  }
+}
+```
+
+Once a VC API exchange URL is acquisitioned from `protocols.vcapi`, a POST
+request is sent with a configuration object (which may be empty) to begin the
+exchange:
+
+```http
+POST /workflows/abc/exchanges/123
+Host: vcapi.example.com
+
+{}
+```
+
+A response will be returned by the exchanger...
+
+If the response object is empty (as above), the exchange is complete and nothing
+is requested from nor offered to the exchange client.
+
+If, however, the object includes `verifiablePresentationRequest`, then the
```

Comment:
Should this link to the [vp-request spec](https://w3c-ccg.github.io/vp-request-spec/#overview)?

Reply:
@BigBlueHat I also wonder if it would be helpful to have a follow-up PR with "FAQ" that can help direct newcomers unfamiliar with VC to the relevant specs. I know I would want more information on how to write or parse these objects.

Reply:
Yeah...this whole site needs reworking into proper documentation...

Reply:
> Should this link to the [vp-request spec](https://w3c-ccg.github.io/vp-request-spec/#overview)?

Fixed in a7bf93d.

The FAQ will have to wait, but I agree it's a needed addition!